server/src/server.ts

```diff
@@ -207,7 +213,14 @@ function openProjectFile(workspacePath: string, fileUri: string): boolean {
                         line.search(GENERATED_YML_DIR_REGEX) === -1
                     );
                 })
-                .map((line) => path.join(path.dirname(fileUri), line))
+                .map((line) =>
+                    isTypeM
+                        ? // In a M type *.kao file, paths are relative to the current *.kao file.
+                          path.join(path.dirname(fileUri), line)
+                        : // In a F type *.kao file, paths are relative to the project's root.
+                          // Here we assume that the root is the workspace path.
+                          path.join(workspacePath, line),
```

Comment:
# If you wish

Duuuude, the `path.join` and the `line` are the same. Consider doing something like…

```ts
from = isTypeM ? path.dirname(fileUri) : workspacePath;
path.join(from, line);
```

Yeah, I’m annoying, but highlighting what changes and what does not changes (+ having a variable name to help) seems nice ^^
Well, actually, I think I’m more disturbed by the fact that your ternary operator spans _five_ freaking lines xD